cmake/Modules/MacroQbtCompilerSettings.cmake

```diff
@@ -1,69 +1,42 @@
+    if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+        add_compile_options(
+            -Wall
+            -Wextra
+            -Wcast-qual
+            -Wcast-align
+            -Winvalid-pch
+            -Woverloaded-virtual
+            -Wold-style-cast
+            -Wnon-virtual-dtor
+            -pedantic
+            -pedantic-errors
         )
 
-        include(CheckCXXCompilerFlag)
-        # check for -pedantic
-        check_cxx_compiler_flag(-pedantic _PEDANTIC_IS_SUPPORTED)
-        if (_PEDANTIC_IS_SUPPORTED)
-            list(APPEND _GCC_COMMON_CXX_FLAGS "-pedantic -pedantic-errors")
-        else (_PEDANTIC_IS_SUPPORTED)
-            list(APPEND _GCC_COMMON_CXX_FLAGS "-Wpedantic")
-        endif (_PEDANTIC_IS_SUPPORTED)
-
-        if (CMAKE_SYSTEM_NAME MATCHES Linux)
-            add_definitions(-D_DEFAULT_SOURCE)
+        if (CMAKE_SYSTEM_NAME MATCHES "Linux")
+            add_compile_definitions(_DEFAULT_SOURCE)
         endif()
 
-        # Clang 5.0 still doesn't support -Wstrict-null-sentinel
+        include(CheckCXXCompilerFlag)
+        # Clang 11 still doesn't support -Wstrict-null-sentinel
         if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
             check_cxx_compiler_flag(-Wstrict-null-sentinel _STRICT_NULL_SENTINEL_IS_SUPPORTED)
             if (_STRICT_NULL_SENTINEL_IS_SUPPORTED)
-                list(APPEND _GCC_COMMON_CXX_FLAGS "-Wstrict-null-sentinel")
-            endif (_STRICT_NULL_SENTINEL_IS_SUPPORTED)
-
-            # Code should be improved to render this not needed
-            list(APPEND _GCC_COMMON_CXX_FLAGS "-Wno-error=unused-function")
-        else ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+                add_compile_options(-Wstrict-null-sentinel)
+            endif()
+        else()
             # GCC supports it
-            list(APPEND _GCC_COMMON_CXX_FLAGS "-Wstrict-null-sentinel")
-        endif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
-
-        string(REPLACE ";" " " _GCC_COMMON_C_AND_CXX_FLAGS_STRING "${_GCC_COMMON_C_AND_CXX_FLAGS}")
-        string(REPLACE ";" " " _GCC_COMMON_CXX_FLAGS_STRING "${_GCC_COMMON_CXX_FLAGS}")
-
-        string(APPEND CMAKE_C_FLAGS " ${_GCC_COMMON_C_AND_CXX_FLAGS_STRING}")
-        string(APPEND CMAKE_CXX_FLAGS " ${_GCC_COMMON_C_AND_CXX_FLAGS_STRING} ${_GCC_COMMON_CXX_FLAGS_STRING}")
-    endif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
-
-    if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
-        set(QBT_ADDITIONAL_FLAGS "/wd4251 /wd4275 /wd4290 /W4" CACHE STRING "Additional qBittorent compile flags")
-    endif ()
-
-    string(APPEND CMAKE_C_FLAGS " ${QBT_ADDITIONAL_FLAGS}")
-    string(APPEND CMAKE_CXX_FLAGS " ${QBT_ADDITIONAL_FLAGS}")
+            add_compile_options(-Wstrict-null-sentinel)
+        endif()
 
-# endif (NOT QBT_ADDITIONAL_FLAGS)
+    elseif (MSVC)
+        add_compile_options(/wd4251 /wd4275 /wd4290 /W4)
```

Comment:
can you add `/wd4127` here 
it produces too much useless noise in the builds 
<details>
```
       ]
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(131): note: see reference to function template instantiation 'T QStringBuilder<QStringBuilder<QString,char>,QString>::convertTo<QString>(void) const' being compiled
        with
        [
            T=QString
        ]
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(131): note: while compiling class template member function 'QStringBuilder<QStringBuilder<QString,char>,QString>::operator QString(void) const'
..\src\gui\mainwindow.cpp(1151): note: see reference to function template instantiation 'QStringBuilder<QStringBuilder<QString,char>,QString>::operator QString(void) const' being compiled
..\src\gui\mainwindow.cpp(1153): note: see reference to class template instantiation 'QStringBuilder<QStringBuilder<QString,char>,QString>' being compiled
C:\Users\Prince\Desktop\Projects\qBittorrent\src\gui\mainwindow.cpp(1298) : warning C4702: unreachable code
[133/298] Building CXX object src\gui\CMakeFiles\qbt_gui.dir\properties\peersadditiondialog.cpp.obj
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qvector.h(796): warning C4127: conditional expression is constant
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qvector.h(796): note: consider using 'if constexpr' statement instead
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qvector.h(789): note: while compiling class template member function 'void QVector<BitTorrent::PeerAddress>::append(const T &)'
        with
        [
            T=BitTorrent::PeerAddress
        ]
..\src\gui\properties\peersadditiondialog.cpp(69): note: see reference to function template instantiation 'void QVector<BitTorrent::PeerAddress>::append(const T &)' being compiled
        with
        [
            T=BitTorrent::PeerAddress
        ]
C:\Users\Prince\Desktop\Projects\qBittorrent\src\gui\properties\peersadditiondialog.h(57): note: see reference to class template instantiation 'QVector<BitTorrent::PeerAddress>' being compiled
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qvector.h(802): warning C4127: conditional expression is constant
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qvector.h(802): note: consider using 'if constexpr' statement instead
[134/298] Building CXX object src\gui\CMakeFiles\qbt_gui.dir\search\searchlistdelegate.cpp.obj
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(120): warning C4127: conditional expression is constant
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(120): note: consider using 'if constexpr' statement instead
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(131): note: see reference to function template instantiation 'T QStringBuilder<QStringBuilder<QLatin1String,QString>,QLatin1String>::convertTo<QString>(void) const' being compiled
        with
        [
            T=QString
        ]
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(131): note: see reference to function template instantiation 'T QStringBuilder<QStringBuilder<QLatin1String,QString>,QLatin1String>::convertTo<QString>(void) const' being compiled
        with
        [
            T=QString
        ]
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(131): note: while compiling class template member function 'QStringBuilder<QStringBuilder<QLatin1String,QString>,QLatin1String>::operator QString(void) const'
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qregularexpression.h(144): note: see reference to function template instantiation 'QStringBuilder<QStringBuilder<QLatin1String,QString>,QLatin1String>::operator QString(void) const' being compiled
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qregularexpression.h(146): note: see reference to class template instantiation 'QStringBuilder<QStringBuilder<QLatin1String,QString>,QLatin1String>' being compiled
[135/298] Building CXX object src\gui\CMakeFiles\qbt_gui.dir\properties\peerlistwidget.cpp.obj
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(120): warning C4127: conditional expression is constant
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(120): note: consider using 'if constexpr' statement instead
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(131): note: see reference to function template instantiation 'T QStringBuilder<QStringBuilder<QString,char>,QString>::convertTo<QString>(void) const' being compiled
        with
        [
            T=QString
        ]
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(131): note: see reference to function template instantiation 'T QStringBuilder<QStringBuilder<QString,char>,QString>::convertTo<QString>(void) const' being compiled
        with
        [
            T=QString
        ]
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(131): note: while compiling class template member function 'QStringBuilder<QStringBuilder<QString,char>,QString>::operator QString(void) const'
..\src\gui\properties\peerlistwidget.cpp(323): note: see reference to function template instantiation 'QStringBuilder<QStringBuilder<QString,char>,QString>::operator QString(void) const' being compiled
..\src\gui\properties\peerlistwidget.cpp(323): note: see reference to class template instantiation 'QStringBuilder<QStringBuilder<QString,char>,QString>' being compiled
[136/298] Building CXX object src\gui\CMakeFiles\qbt_gui.dir\search\searchsortmodel.cpp.obj
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(120): warning C4127: conditional expression is constant
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(120): note: consider using 'if constexpr' statement instead
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(131): note: see reference to function template instantiation 'T QStringBuilder<QStringBuilder<QLatin1String,QString>,QLatin1String>::convertTo<QString>(void) const' being compiled
        with
        [
            T=QString
        ]
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(131): note: see reference to function template instantiation 'T QStringBuilder<QStringBuilder<QLatin1String,QString>,QLatin1String>::convertTo<QString>(void) const' being compiled
        with
        [
            T=QString
        ]
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(131): note: while compiling class template member function 'QStringBuilder<QStringBuilder<QLatin1String,QString>,QLatin1String>::operator QString(void) const'
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qregularexpression.h(144): note: see reference to function template instantiation 'QStringBuilder<QStringBuilder<QLatin1String,QString>,QLatin1String>::operator QString(void) const' being compiled
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qregularexpression.h(146): note: see reference to class template instantiation 'QStringBuilder<QStringBuilder<QLatin1String,QString>,QLatin1String>' being compiled
[137/298] Building CXX object src\gui\CMakeFiles\qbt_gui.dir\optionsdialog.cpp.obj
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(120): warning C4127: conditional expression is constant
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(120): note: consider using 'if constexpr' statement instead
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(131): note: see reference to function template instantiation 'T QStringBuilder<QString,QLatin1String>::convertTo<QString>(void) const' being compiled
        with
        [
            T=QString
        ]
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(131): note: see reference to function template instantiation 'T QStringBuilder<QString,QLatin1String>::convertTo<QString>(void) const' being compiled
        with
        [
            T=QString
        ]
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(131): note: while compiling class template member function 'QStringBuilder<QString,QLatin1String>::operator QString(void) const'
..\src\gui\optionsdialog.cpp(470): note: see reference to function template instantiation 'QStringBuilder<QString,QLatin1String>::operator QString(void) const' being compiled
..\src\gui\optionsdialog.cpp(470): note: see reference to class template instantiation 'QStringBuilder<QString,QLatin1String>' being compiled
[138/298] Building CXX object src\gui\CMakeFiles\qbt_gui.dir\search\pluginselectdialog.cpp.obj
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qvector.h(796): warning C4127: conditional expression is constant
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qvector.h(796): note: consider using 'if constexpr' statement instead
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qvector.h(789): note: while compiling class template member function 'void QVector<QTreeWidgetItem *>::append(const T &)'
        with
        [
            T=QTreeWidgetItem *
        ]
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qvector.h(293): note: see reference to function template instantiation 'void QVector<QTreeWidgetItem *>::append(const T &)' being compiled
        with
        [
            T=QTreeWidgetItem *
        ]
..\src\gui\search\pluginselectdialog.cpp(242): note: see reference to class template instantiation 'QVector<QTreeWidgetItem *>' being compiled
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qvector.h(802): warning C4127: conditional expression is constant
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qvector.h(802): note: consider using 'if constexpr' statement instead
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(120): warning C4127: conditional expression is constant
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(120): note: consider using 'if constexpr' statement instead
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(131): note: see reference to function template instantiation 'T QStringBuilder<QString,QLatin1String>::convertTo<QString>(void) const' being compiled
        with
        [
            T=QString
        ]
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(131): note: see reference to function template instantiation 'T QStringBuilder<QString,QLatin1String>::convertTo<QString>(void) const' being compiled
        with
        [
            T=QString
        ]
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(131): note: while compiling class template member function 'QStringBuilder<QString,QLatin1String>::operator QString(void) const'
..\src\gui\search\pluginselectdialog.cpp(363): note: see reference to function template instantiation 'QStringBuilder<QString,QLatin1String>::operator QString(void) const' being compiled
..\src\gui\search\pluginselectdialog.cpp(366): note: see reference to class template instantiation 'QStringBuilder<QString,QLatin1String>' being compiled
[139/298] Building CXX object src\gui\CMakeFiles\qbt_gui.dir\properties\propertieswidget.cpp.obj
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(120): warning C4127: conditional expression is constant
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(120): note: consider using 'if constexpr' statement instead
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(131): note: see reference to function template instantiation 'T QStringBuilder<QStringBuilder<QString,char>,QString>::convertTo<QString>(void) const' being compiled
        with
        [
            T=QString
        ]
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(131): note: see reference to function template instantiation 'T QStringBuilder<QStringBuilder<QString,char>,QString>::convertTo<QString>(void) const' being compiled
        with
        [
            T=QString
        ]
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(131): note: while compiling class template member function 'QStringBuilder<QStringBuilder<QString,char>,QString>::operator QString(void) const'
..\src\gui\properties\propertieswidget.cpp(380): note: see reference to function template instantiation 'QStringBuilder<QStringBuilder<QString,char>,QString>::operator QString(void) const' being compiled
..\src\gui\properties\propertieswidget.cpp(380): note: see reference to class template instantiation 'QStringBuilder<QStringBuilder<QString,char>,QString>' being compiled
[140/298] Building CXX object src\gui\CMakeFiles\qbt_gui.dir\shutdownconfirmdialog.cpp.obj
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(120): warning C4127: conditional expression is constant
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(120): note: consider using 'if constexpr' statement instead
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(131): note: see reference to function template instantiation 'T QStringBuilder<QString,char>::convertTo<QString>(void) const' being compiled
        with
        [
            T=QString
        ]
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(131): note: see reference to function template instantiation 'T QStringBuilder<QString,char>::convertTo<QString>(void) const' being compiled
        with
        [
            T=QString
        ]
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(131): note: while compiling class template member function 'QStringBuilder<QString,char>::operator QString(void) const'
..\src\gui\shutdownconfirmdialog.cpp(139): note: see reference to function template instantiation 'QStringBuilder<QString,char>::operator QString(void) const' being compiled
..\src\gui\shutdownconfirmdialog.cpp(139): note: see reference to class template instantiation 'QStringBuilder<QString,char>' being compiled
[141/298] Building CXX object src\gui\CMakeFiles\qbt_gui.dir\search\searchwidget.cpp.obj
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(120): warning C4127: conditional expression is constant
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(120): note: consider using 'if constexpr' statement instead
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(131): note: see reference to function template instantiation 'T QStringBuilder<QStringBuilder<QLatin1String,QString>,QLatin1String>::convertTo<QString>(void) const' being compiled
        with
        [
            T=QString
        ]
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(131): note: see reference to function template instantiation 'T QStringBuilder<QStringBuilder<QLatin1String,QString>,QLatin1String>::convertTo<QString>(void) const' being compiled
        with
        [
            T=QString
        ]
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(131): note: while compiling class template member function 'QStringBuilder<QStringBuilder<QLatin1String,QString>,QLatin1String>::operator QString(void) const'
C:\Qt\5.14.1\msvc2017_64\include\QtCore\qregularexpression.h(144): note: see reference to function template instantiation 'QStringBuilder<QStringBuilder<QLatin1String,QString>,QLatin1String>::operator QString(void) const' being compiled
C:\Qt\5.14.1\msvc2017_64\include\QtCore\qregularexpression.h(146): note: see reference to class template instantiation 'QStringBuilder<QStringBuilder<QLatin1String,QString>,QLatin1String>' being compiled
[143/298] Building CXX object src\gui\CMakeFiles\qbt_gui.dir\properties\speedwidget.cpp.obj
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(120): warning C4127: conditional expression is constant
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(120): note: consider using 'if constexpr' statement instead
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(131): note: see reference to function template instantiation 'T QStringBuilder<QStringBuilder<QLatin1String,QString>,QLatin1String>::convertTo<QString>(void) const' being compiled
        with
        [
            T=QString
        ]
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(131): note: see reference to function template instantiation 'T QStringBuilder<QStringBuilder<QLatin1String,QString>,QLatin1String>::convertTo<QString>(void) const' being compiled
        with
        [
            T=QString
        ]
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(131): note: while compiling class template member function 'QStringBuilder<QStringBuilder<QLatin1String,QString>,QLatin1String>::operator QString(void) const'
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qregularexpression.h(144): note: see reference to function template instantiation 'QStringBuilder<QStringBuilder<QLatin1String,QString>,QLatin1String>::operator QString(void) const' being compiled
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qregularexpression.h(146): note: see reference to class template instantiation 'QStringBuilder<QStringBuilder<QLatin1String,QString>,QLatin1String>' being compiled
[144/298] Building CXX object src\gui\CMakeFiles\qbt_gui.dir\tagfilterproxymodel.cpp.obj
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(120): warning C4127: conditional expression is constant
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(120): note: consider using 'if constexpr' statement instead
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(131): note: see reference to function template instantiation 'T QStringBuilder<QStringBuilder<QLatin1String,QString>,QLatin1String>::convertTo<QString>(void) const' being compiled
        with
        [
            T=QString
        ]
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(131): note: see reference to function template instantiation 'T QStringBuilder<QStringBuilder<QLatin1String,QString>,QLatin1String>::convertTo<QString>(void) const' being compiled
        with
        [
            T=QString
        ]
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(131): note: while compiling class template member function 'QStringBuilder<QStringBuilder<QLatin1String,QString>,QLatin1String>::operator QString(void) const'
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qregularexpression.h(144): note: see reference to function template instantiation 'QStringBuilder<QStringBuilder<QLatin1String,QString>,QLatin1String>::operator QString(void) const' being compiled
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qregularexpression.h(146): note: see reference to class template instantiation 'QStringBuilder<QStringBuilder<QLatin1String,QString>,QLatin1String>' being compiled
[145/298] Building CXX object src\gui\CMakeFiles\qbt_gui.dir\speedlimitdialog.cpp.obj
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(120): warning C4127: conditional expression is constant
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(120): note: consider using 'if constexpr' statement instead
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(131): note: see reference to function template instantiation 'T QStringBuilder<QStringBuilder<QLatin1String,QString>,QLatin1String>::convertTo<QString>(void) const' being compiled
        with
        [
            T=QString
        ]
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(131): note: see reference to function template instantiation 'T QStringBuilder<QStringBuilder<QLatin1String,QString>,QLatin1String>::convertTo<QString>(void) const' being compiled
        with
        [
            T=QString
        ]
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qstringbuilder.h(131): note: while compiling class template member function 'QStringBuilder<QStringBuilder<QLatin1String,QString>,QLatin1String>::operator QString(void) const'
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qregularexpression.h(144): note: see reference to function template instantiation 'QStringBuilder<QStringBuilder<QLatin1String,QString>,QLatin1String>::operator QString(void) const' being compiled
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qregularexpression.h(146): note: see reference to class template instantiation 'QStringBuilder<QStringBuilder<QLatin1String,QString>,QLatin1String>' being compiled
[146/298] Building CXX object src\gui\CMakeFiles\qbt_gui.dir\properties\trackerlistwidget.cpp.obj
..\src\gui\properties\trackerlistwidget.cpp(366): warning C4458: declaration of 'data' hides class member
C:\Qt\5.14.1\msvc2017_64\include\QtWidgets/qwidget.h(747): note: see declaration of 'QWidget::data'
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qvector.h(796): warning C4127: conditional expression is constant
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qvector.h(796): note: consider using 'if constexpr' statement instead
C:\Qt\5.14.1\msvc2017_64\include\QtCore/qvector.h(789): note: while compiling class template member function 'void QVector<T>::append(const T &)'
```
</details>

Reply:
@jagannatharjun 
I think it is sensible to have both build systems using the same flags, i.e. you might want to submit a PR changing the qmake part first so that this PR can follow.

Reply:
qmake uses /W3, which have c4127 already disable 

so may be then make it /W3 then.

Reply:
I think it is caused by `/W4` which isn't present in qmake build. I would suggest to remove it and rely on the default.

Reply:
>so maybe make it /W3 then.

Agreed. MSVC IDE defaults to `/W3` too. 
https://docs.microsoft.com/en-us/cpp/build/reference/compiler-option-warning-level?view=vs-2019

Reply:
@Chocobo1 @jagannatharjun 

Hmm, I have set it to `/W3` for now, but looking through https://docs.microsoft.com/en-us/cpp/error-messages/compiler-warnings/compiler-warnings-c4000-through-c4199 and https://docs.microsoft.com/en-us/cpp/preprocessor/compiler-warnings-that-are-off-by-default, there seem to be some level 4 warnings that would be nice to have when building on Windows, such as https://docs.microsoft.com/en-us/cpp/error-messages/compiler-warnings/compiler-warning-level-4-c4189.

I think that ideally, we'd use `/W4 /Wall`, and then blacklist those that produce too much noise. However, _where_ the noise is coming from is also important. For example, I'd actually like to have warning `C4127`, but only for our code - is there a way to tell that to the compiler? e.g. "if this warning comes from Qt code, ignore it"?

Reply:
I'll consider this resolved for now. Apparently what we want is MSVC's `external` include directives. We can later discuss if we really benefit from `/W4` or `/Wall`. I should note that at the time of writing, CMake does not natively support this MSVC feature. That is being tracked here: https://gitlab.kitware.com/cmake/cmake/-/issues/17904